problems/string-format/string-format.rs

```diff
@@ -0,0 +1,18 @@
+fn format(orig_str: &str, args: Vec<&str>) -> String {
```

Comment:
Sounds like a nice opportunity to look into macros, specifically the [`format!`](https://doc.rust-lang.org/src/alloc/macros.rs.html#105-107) macro?